clconf/template.go

```diff
@@ -0,0 +1,198 @@
+package clconf
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"syscall"
+)
+
+const setuid = 4
+const setgid = 2
+const sticky = 1
+
+// Settings for ProcessTemplates.
+type TemplateSettings struct {
+	Flatten           bool
+	KeepExistingPerms bool
+	Rm                bool
+	FileMode          string
+	DirMode           string
+	Extension         string
+}
+
+type pathWithRelative struct {
+	fullPath string
+	relPath  string
+}
+
+// ProcessTemplates processes templates.
```

Comment:
This is the API contract.  As such, its documentation should include more detail.